packages/plugins/opentelemetry/src/plugin.ts

```diff
@@ -0,0 +1,291 @@
+            [SEMATTRS_HTTP_CLIENT_IP]: ips?.split(',')[0],
+            [SEMATTRS_HTTP_SERVER_NAME]: serviceName,
+            [SEMATTRS_HTTP_USER_AGENT]: userAgent,
+          },
+          kind: SpanKind.SERVER,
+        },
+        rootContext,
+      );
+
+      requestContextMapping.set(request, trace.setSpan(rootContext, rootSpan));
+    },
+    onParse(onParsePayload) {
+      const shouldTracePrase =
+        typeof options.spans?.graphqlParse === 'function'
+          ? options.spans.graphqlParse(onParsePayload)
+          : (options.spans?.graphqlParse ?? true);
+
+      const { context } = onParsePayload;
+
+      if (shouldTracePrase && requestContextMapping.has(context.request)) {
+        const rootContext = requestContextMapping.get(context.request)!;
+        const parseSpan = tracer.startSpan(
+          'graphql.parse',
+          {
+            attributes: {
+              [SEMATTRS_GRAPHQL_DOCUMENT]: context.params.query,
+              [SEMATTRS_GRAPHQL_OPERATION_NAME]: context.params.operationName,
+            },
+            kind: SpanKind.INTERNAL,
+          },
+          rootContext,
+        );
+
+        return ({ result }) => {
+          if (result instanceof Error) {
+            parseSpan.setStatus({
+              code: SpanStatusCode.ERROR,
+              message: result.message,
+            });
+          }
+
+          parseSpan.end();
+        };
+      }
+    },
+    onExecute({ args }) {
+      const shouldTraceExecute = options.spans?.graphqlExecute ?? true;
+      if (requestContextMapping.has(args.contextValue.request)) {
+        const rootContext = requestContextMapping.get(args.contextValue.request)!;
+        const operation = getOperationAST(args.document, args.operationName);
+        const executeSpan = tracer.startSpan(
+          'graphql.execute',
+          {
+            attributes: {
+              [SEMATTRS_GRAPHQL_OPERATION_TYPE]: operation.operation,
+              [SEMATTRS_GRAPHQL_OPERATION_NAME]: args.operationName,
+              [SEMATTRS_GRAPHQL_DOCUMENT]: print(args.document),
+            },
+            kind: SpanKind.INTERNAL,
+          },
+          rootContext,
+        );
+
+        return {
+          onExecuteDone({ result }) {
+            if (!isAsyncIterable(result)) {
+              if (result.errors && result.errors.length > 0) {
+                executeSpan.setAttribute(SEMATTRS_GRAPHQL_ERROR_COUNT, result.errors.length);
+                executeSpan.setStatus({
+                  code: SpanStatusCode.ERROR,
+                  message: result.errors.map(e => e.message).join(', '),
+                });
+
+                for (const error in result.errors) {
+                  executeSpan.recordException(error);
+                }
+              }
+            }
+
+            executeSpan.end();
+          },
+        };
+      }
+    },
+    onFetch({ context, options, url, setOptions, executionRequest }) {
+      if (requestContextMapping.has(context.request)) {
```

Comment:
```suggestion
      const otelContext = requestContextMapping.get(context.request);
      if (otelContext) {
```
This would avoid `!` in the next line

Reply:
fixed